portal/app/[locale]/staking-dashboard/_components/stakeTable/lockupTime.tsx

```diff
@@ -1,15 +1,46 @@
 import { DurationTime } from 'components/durationTime'
+import { useTranslations } from 'next-intl'
+import Skeleton from 'react-loading-skeleton'
+import { formatNumber } from 'utils/format'
+
+import { useCalculateApy } from '../../_hooks/useCalculateApy'
 
 type Props = {
   lockupTime: bigint
+  tokenId: string
 }
 
-export const LockupTime = function ({ lockupTime }: Props) {
+export const LockupTime = function ({ lockupTime, tokenId }: Props) {
+  const t = useTranslations('staking-dashboard.table')
   const seconds = Number(lockupTime)
+  const {
+    data: apy,
+    error,
+    isLoading,
+  } = useCalculateApy({ tokenId: BigInt(tokenId) })
+
+  const renderApy = function () {
+    if (isLoading) {
+      return <Skeleton className="h-4 w-16" />
+    }
+
+    if (error || apy === undefined) {
+      return <span className="text-xs font-normal text-neutral-500">-</span>
+    }
+
+    return (
+      <span className="text-xs font-normal text-emerald-600">
+        {t('apy', { percentage: formatNumber(apy) })}
+      </span>
+    )
+  }
 
   return (
-    <span className="text-neutral-500">
-      <DurationTime seconds={seconds} />
-    </span>
+    <div className="flex flex-col">
+      <span className="text-neutral-500">
+        <DurationTime seconds={seconds} />
+      </span>
+      {renderApy()}
```

Comment:
Would this render APY for burned tokens? I think it will. Perhaps we want to use 2 different columns depending on the filter being `active` or `withdrawn`. Is this correct, or am I missing something?

Reply:
Once burned, I don't think there's any APY to show, as rewards wouldn't be "earned" anymore